src/mem_mng.c

```diff
@@ -88,7 +82,22 @@ manager_free_mem_alloc(struct Compartment* comp, void* ptr)
     }
     size_t to_return = curr_alloc->size;
     free(curr_alloc);
-    // TODO subtract allocated memory
 
     return to_return;
 }
+
+struct mem_alloc*
+get_alloc_struct_from_ptr(struct Compartment* comp, uintptr_t ptr)
```

Comment:
Let's add a doc string (here or at the declaration).

Reply:
Added in https://github.com/capablevms/CHERI-ELF-comp/pull/3/commits/a8e8e0718c2257e9931f3ddc74bfd9915b4a1010.